samples/bluetooth/central_ht/src/main.c

```diff
@@ -0,0 +1,277 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2020 SixOctets Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <stddef.h>
+#include <errno.h>
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <sys/byteorder.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+
+static struct bt_conn *default_conn;
+static struct bt_uuid_16 uuid = BT_UUID_INIT_16(0);
+static struct bt_gatt_discover_params discover_params;
+static struct bt_gatt_subscribe_params subscribe_params;
+
+static double power(double x, s8_t y)
+{
+	double result = 1;
+
+	if (y < 0) {
+		y = -y;
+		while (y--) {
+			result /= x;
+		}
+	} else {
+		while (y--) {
+			result *= x;
+		}
+	}
+
+	return result;
+}
+
+static u8_t notify_func(struct bt_conn *conn,
+			   struct bt_gatt_subscribe_params *params,
+			   const void *data, u16_t length)
+{
+	double temperature;
+	u32_t mantissa;
+	s8_t exponent;
+	u32_t *ieee;
+
+	if (!data) {
+		printk("[UNSUBSCRIBED]\n");
+		params->value_handle = 0U;
+		return BT_GATT_ITER_STOP;
+	}
+	ieee = (u32_t *)&((u8_t *)data)[1];
```

Comment:
I think we should create helper functions for this. Could you add that?
bt_float convert functions would be nice.

Reply:
Additionally this looks like unaligned access, so it should be using `UNALIGNED_GET()` as far as I see

Reply:
Might be good to enable unaligned fault to make sure that it is done correctly.
```
SCB->CCR |= SCB_CCR_UNALIGN_TRP_Msk;
```

Reply:
I used sys_pt_le24 function and it is working fine

Reply:
I used sys _put and sys_get function for unaligned access.